### PR TITLE
MCR-2159 separated methods to retrieve MCRObjectIDs from pathParams

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2159).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212&bid=16856521)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java [hunk 1]
```diff
@@ -810,7 +818,11 @@ private static MCRRestAPISortObject createSortObject(String input) throws MCRRes
         return result;
     }
 
-    private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIException {
+    //TODO move into generic package for v1 and v2 REST-API
+    public static MCRObjectID retrieveMCRObjectID(String idString) throws MCRRestAPIException {
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid reassigning parameters such as 'idString'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java [hunk 2]
```diff
@@ -875,18 +887,22 @@ private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIExc
         }
     }
 
-    private static MCRDerivate retrieveMCRDerivate(MCRObjectID parentObjId, String derIDString)
+    //TODO move into generic package for v1 and v2 REST-API
+    public static MCRObjectID retrieveMCRDerivateID(MCRObjectID parentObjId, String derIDString)
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid reassigning parameters such as 'derIDString'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java [hunk 3]
```diff
@@ -521,7 +525,7 @@ public static Response listObjects(UriInfo info, String format, String filter,
      * 
      * @see MCRRestAPIObjects#listDerivates(UriInfo, String, String, String)
      */
-    public static Response listDerivates(UriInfo info, String mcrObjID, String format,
+    public static Response listDerivates(UriInfo info, String mcrId, String format,
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid really long methods.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java [hunk 4]
```diff
@@ -875,18 +887,22 @@ private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIExc
...
+            return null;
+        }
 
         String derKey = "mcr"; // the default value for the key
         if (derIDString.contains(":")) {
             int pos = derIDString.indexOf(":");
             derKey = derIDString.substring(0, pos);
             derIDString = derIDString.substring(pos + 1);
-            if (!derKey.equals("mcr") && !derKey.equals("label")) {
+            if (!derKey.equals("mcr") && !derKey.equals("label") && !derKey.equals("classification")) {
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 2 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [The method 'retrieveMCRDerivateID(MCRObjectID, String)' has an NPath complexity of 8100](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 3 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [The method 'listDerivates(UriInfo, String, String, String)' has an NPath complexity of 3240](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d0fc77c24f70ca10c79250f2f38cf3a298c2636c! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5292-pull_request-921

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212&bid=16856521)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 1 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid reassigning parameters such as 'idString'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 3 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [The method 'listDerivates(UriInfo, String, String, String)' has an NPath complexity of 3240](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 2 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [The method 'retrieveMCRDerivateID(MCRObjectID, String)' has an NPath complexity of 8100](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 3 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid really long methods.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 2 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Avoid reassigning parameters such as 'derIDString'](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 4 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212&bid=16856521)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 3 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java:528]: Codacy Issue found: [The method 'listDerivates(UriInfo, String, String, String)' has an NPath complexity of 3240](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java [hunk 5]
```diff
@@ -875,18 +888,23 @@ private static MCRObject retrieveMCRObject(String idString) throws MCRRestAPIExc
         }
     }
 
-    private static MCRDerivate retrieveMCRDerivate(MCRObjectID parentObjId, String derIDString)
+    //TODO move into generic package for v1 and v2 REST-API
+    public static MCRObjectID retrieveMCRDerivateID(MCRObjectID parentObjId, String paramDerID)
```
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java:892]: Codacy Issue found: [The method 'retrieveMCRDerivateID(MCRObjectID, String)' has an NPath complexity of 8100](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java (hunk 3 above)
Comment (adlerre) [inline mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIObjectsHelper.java:528]: Codacy Issue found: [Avoid really long methods.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5129212)

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRRestAuthorizationFilter.java [hunk 6]
```diff
@@ -93,14 +95,20 @@ private void checkRestAPIAccess(ContainerRequestContext requestContext, MCRRestA
...
-            .map(Optional::of)
-            .orElseGet(() -> Optional.ofNullable(objectId));
+        String objectId, String derId, String path) throws MCRRestAPIException {
+        Optional<String> optObjId = Optional.ofNullable(MCRRestAPIObjectsHelper.retrieveMCRObjectID(objectId))
+            .map(MCRObjectID::toString);
+        Optional<String> optDerId = Optional.empty();
+        if (optObjId.isPresent()) {
+            optDerId = Optional
+                .ofNullable(
+                    MCRRestAPIObjectsHelper.retrieveMCRDerivateID(MCRObjectID.getInstance(optObjId.get()), derId))
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/MCRRestAuthorizationFilter.java:105]: Write a [PreMatching Filter](https://www.logicbig.com/tutorials/java-ee-tutorial/jax-rs/filters-pre-matching.html) and register it within `org.mycore.restapi.v1.MCRRestV1App` to rewrite the URL on-the-fly without tighter coupling (read: leave `MCRRestAuthorizationFilter` untouched)

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by rsteph-de** [comment]
will use Jersey prematching filter as suggested

**Pull request closed without merging**